force-app/main/default/classes/SOQL.cls

```diff
@@ -16,343 +16,463 @@
+                    FROM Opportunity
+                    WHERE AccountId = :accountId
+                    AND StageName = 'Closed Won'
+                    AND Amount != null
+                    LIMIT 50000];
+        } catch(Exception e) {
+            return new List<Opportunity>();
+        }
+    }
+
+    /**
+     * Question 5
+     * Returns the total count of Account records
+     * @return Integer The count of Account records
+     */
+    public static Integer getAccountCount() {
+        try {
+            Integer count = [SELECT COUNT() FROM Account WHERE Name != null];
+            return count != null ? count : 0;
+        } catch(Exception e) {
+            return 0;
+        }
+    }
+
+    /**
+     * Question 6
+     * Returns tasks due within next 7 days
+     * @return Map<Id, Task> Map of Task Ids to Task records
+     */
+    public static Map<Id, Task> getTasksDueNextWeek() {
+        try {
+            Date today = Date.today();
+            Date nextWeek = today.addDays(7);
+
+            List<Task> tasks = [SELECT Id, Subject, ActivityDate, Status
+                               FROM Task
+                               WHERE ActivityDate >= :today
+                               AND ActivityDate <= :nextWeek
+                               AND Subject != null
+                               LIMIT 50000];
+            return new Map<Id, Task>(tasks);
+        } catch(Exception e) {
+            return new Map<Id, Task>();
+        }
+    }
+
+    /**
+     * Question 7
+     * Returns contacts with example.com email domain
+     * @return List<Contact> List of filtered contacts
+     */
+    public static List<Contact> getContactsByEmailDomain() {
+        try {
+            return [SELECT Id, Name, Email
+                    FROM Contact
+                    WHERE Email LIKE '%example.com'
+                    AND Email != null
+                    LIMIT 50000];
+        } catch(Exception e) {
+            return new List<Contact>();
+        }
+    }
+
+    /**
+     * Question 8
+     * Returns opportunities containing specific keyword in name
+     * @param keyword The keyword to search in Opportunity Name
+     * @return List<Opportunity> List of filtered opportunities
+     */
+    public static List<Opportunity> getOpportunitiesByNameKeyword(String keyword) {
+        if (String.isBlank(keyword)) {
+            return new List<Opportunity>();
+        }
+
+        String searchTerm = '%' + keyword + '%';
+        return [SELECT Id, Name
+                FROM Opportunity
+                WHERE Name LIKE :searchTerm
+                LIMIT 50000];
+    }
+
+    /**
+     * Question 9
+     * Returns cases with specific origin criteria
+     * @return List<Case> List of filtered cases
+     */
+    public static List<Case> getFilterCaseOrigins() {
+        return [SELECT Id, Origin
+                FROM Case
+                WHERE Origin NOT IN ('Web', 'Email')
+                AND Origin != null
```

Comment:
This one gets a lot of people but null is not the same as blank or empty. Try to use the isEmpty() or ' '